chia/consensus/blockchain.py

```diff
@@ -378,14 +379,15 @@ async def add_block(
         if not genesis and prev_block is not None:
             self.add_block_record(prev_block)
 
-        block_record = block_to_block_record(
-            self.constants,
-            self,
-            required_iters,
-            block,
-            sub_slot_iters=sub_slot_iters,
-            prev_ses_block=prev_ses_block,
-        )
+        if block_record is None:
+            block_record = block_to_block_record(
```

Comment:
does this still happen? if it's a subtle path, it might make sense to add a comment to the code explaining when we end up here

Reply:
this will happen when we get here from full_node.add_block call